### PR TITLE
Ft-change-how-user-adds-meal-15711085 and Ft-upgrade-nutrition-diary

### PR DESCRIPTION
**Pull request by Kalela**
# What does this PR do?
- Combines the meal time and the meal ingredient forms into one form. 
- Separates the meal table into a planned meal and consumed meal. The planned meal table holds the nutrition plan meal and the consumed meal contains the actual meal that the user ate at that time.

#### Old method of meal input
screen shot 2018-05-15 at 17 15 54
 New method of meal input
screen shot 2018-05-28 at 18 32 39

#### Planned and Consumed tables
screen shot 2018-05-29 at 12 01 19

### How should this be manually tested?
- Set up wger locally as shown in the README.
- Log in with username ```admin``` and password ```admin```.
- Go to the nutrition/nutrition plans tab.
- Add a nutrition plan
- Add a planned meal in the plan.
- Add a consumed meal in the plan.

### What are the relevant pivotal tracker stories?
- Change the way the user add meals 
    #157110853
- Upgrade the nutrition diary feature
   #157110849

### CODE REVIEW

**Review by meshnesh — CHANGES_REQUESTED** [review]
fix all the conflicting files

File: extras/bench/test_Simple.py [hunk 1]
```diff
@@ -3,9 +3,10 @@
 Simple FunkLoad test
 '''
 import unittest
-from random import random
+from random import random # noqa
```
Comment (betsybeth) [inline extras/bench/test_Simple.py]: What is noqa?

**Comment by mutoromike** [comment]
What do you think about fixing the builds before we could merge...?

File: extras/bench/test_Simple.py (hunk 1 above)
Comment (Kalela) [reply extras/bench/test_Simple.py]: #noqa is a keyword for flake8 functionality. It makes flake8 skip the line. Preventing flake* from flagging errors where it shouldn't.

**Comment by Kalela** [comment]
Yes...will work on fixing the build.

**Review by FelixWambiri — CHANGES_REQUESTED** [review]
It would be ideal if you fixed the failing tests and merge conflicts first.

File: wger/core/templates/base.html [hunk 2]
```diff
@@ -65,6 +68,21 @@ <h2 id="page-title" class="page-header">
         {% block sidebar %}
         {% endblock %}
     </div>
-   
+    <div class="col-sm-6" id="meal-content">
+        {% if messages and not no_messages %}
+            {% for message in messages %}
```
Comment (frankip) [inline wger/core/templates/base.html]: good job

Comment (frankip) [inline wger/core/templates/base.html:74]: great work

**Review by lakivisi-zz — CHANGES_REQUESTED** [review]
COuld you make this PR's 2 small ones instead of one big one

**Review by betsybeth — CHANGES_REQUESTED** [review]
Fix the conflicting files before fixing the travis build so that you wont have a hard time building travis

File: wger/core/templates/base.html [hunk 3]
```diff
@@ -24,15 +24,18 @@ <h2 id="page-title" class="page-header">
             {% block title %}{% endblock %}
         </h2>
     </div>
-    <div class="col-sm-4" style="padding-top: 15px; padding-bottom: 15px;">
-        <div class="btn-group">
+    <div class="col-sm-3">
```
Comment (FelixWambiri) [inline wger/core/templates/base.html:27]: Nice work on the frontend.

File: wger/exercises/tests/test_exercise.py [hunk 4]
```diff
@@ -512,8 +512,8 @@ def test_overview_cache_update(self):
 
         old_exercise_bg = cache\
             .get(cache_mapper.get_exercise_muscle_bg_key(2))
```
Comment (FelixWambiri) [inline wger/exercises/tests/test_exercise.py:514]: Nice work on adding tests

File: wger/nutrition/templates/meal/new_meal_create.html [hunk 5]
```diff
@@ -0,0 +1,48 @@
+{% extends "base.html" %}
+ {% load i18n %}
+ {% load staticfiles %}
+ {% load wger_extras %}
+
```
Comment (FelixWambiri) [inline wger/nutrition/templates/meal/new_meal_create.html]: Nice work on using javascript in this file

**Review by FelixWambiri — COMMENTED** [review]
Nice implementation. Please fix the tests and make sure that they are passing

**Review by meshnesh — APPROVED** [review]

**Review by lakivisi-zz — CHANGES_REQUESTED** [review]
Merge develop

**Review by FelixWambiri — APPROVED** [review]
Nice work on finishing the task. On the two tables, it would look better if you tried placing them on the same level.

**Review by mutoromike — APPROVED** [review]

**Review by betsybeth — APPROVED** [review]

**Comment by Kalela** [comment]
@FelixWambiri  Working on it

**Comment by Kalela** [comment]
@FelixWambiri what do you think of how it looks now?

**Review by frankip — APPROVED** [review]
👍

**Review by meshnesh — CHANGES_REQUESTED** [review]
fix the conflicting files, and don't push your settings.py file online

**Review by meshnesh — APPROVED** [review]

**Comment by Kalela** [comment]
Which conflicting files????????

**Pull request merged**